Fix for `!important` not being carried from mixing to custom property application

nazar-pc opened the pull request:
Fixes #158

TimvdLippe commented on tests/apply-shim.html:
```diff
@@ -273,6 +289,19 @@
       });
     });
+    suite('!important', function() {
+      var style, ast; // eslint-disable-line no-unused-vars
```
Style can be deleted here

TimvdLippe commented on tests/apply-shim.html:
```diff
@@ -138,6 +138,22 @@
+    <style>
+    :host {
+      --mixin-important: {
+        border: 2px solid black !important;
```
Could you add another property which does not have important in both mixins and assert it takes the value of the last one?

nazar-pc commented:
Addressed comments in last commit

TimvdLippe commented on tests/apply-shim.html:
```diff
@@ -273,6 +290,20 @@
       });
     });
+    suite('!important', function() {
+      var ast; // eslint-disable-line no-unused-vars
```
Comment can now be removed

TimvdLippe commented on tests/apply-shim.html:
```diff
@@ -138,6 +138,23 @@
+        background-color: white;
+      };
+      --mixin: {
+        border: 1px dotted orange;
```
Please also add a `background-color: red;` here

TimvdLippe replied:
Could you also add a case where both properties have `!important` and assert the last one is actually used?

TimvdLippe commented on tests/apply-shim.html:
```diff
@@ -273,6 +290,20 @@
+        var application = ast.rules[1];
+        assert.match(application.cssText, /^border:\s*var\(--mixin-important_-_border\)\s*!important/);
+        assert.notMatch(application.cssText, /background-color:\s*var\(--mixin-important_-_background-color\)\s*!important/);
+        assert.match(application.cssText, /border:\s*var\(--mixin_-_border,\s*var\(--mixin-important_-_border\)\)/);
```
Maybe split this cases up so that debugging of the test cases becomes easier. For example:

1. `!important in earlier selector correctly translates to !important in mixin properties`
1. `Properties without !important in a mixin with !important are treated normally`
1. `two mixins with both !important are treated as regular ordered selectors`
1. `fallback of unrelated property are untouched`

nazar-pc commented:
Added even more test cases as requested.
Not sure why it fails on Travis. Some unrelated tests also failed for me locally even prior to any changes. Added tests however do not fail both in Chromium Nightly and in Firefox Nightly. I don't have Edge or Safari on Linux to check those.

TimvdLippe commented:
Hm that is weird, as these tests appear to fail in both Chrome and Firefox.

nazar-pc commented:
I suspect they fail because `apply-shim.min.js` is not re-generated before running tests (I didn't commit updated minified version, but test cases for whatever reason call minified version).

TimvdLippe commented:
Ah that would make sense. Could you regenerate it and retry?

nazar-pc commented:
Do you mean commit generated file?

TimvdLippe commented:
Yes please. Other PRs have that as well

nazar-pc commented:
It doesn't seem right as it would cause merge conflicts. Anyway, added in last commit.

TimvdLippe commented:
@nazar-pc Thanks for the fix!

nazar-pc commented:
Thanks for quick review! Now hoping for new release on NPM in near future 😀
